src/seedu/addressbook/commands/Command.java

```diff
@@ -63,11 +63,20 @@ protected ReadOnlyPerson getTargetPerson() throws IndexOutOfBoundsException {
         return relevantPersons.get(getTargetIndex() - DISPLAYED_INDEX_OFFSET);
     }
 
+    // returns false, by default for commands that are mutating
+
+    public boolean isMutating() {
+        return false;
```

Comment:
As mentioned during the tutorial, it may not be a good idea to have a default implementation for `isMutating` as it is uncertain as to what is the default behaviour for a "generic command".

Since `Command` is an abstract class, we can make `isMutating` abstract and let the subclasses provide the method's implementation.